Small changes to gossip related telemetry docs (#9846)
hashicorp/consul: website/content/docs/agent/telemetry.mdx

| `consul.memberlist.tcp.connect`       | This metric counts the number of times an agent has initiated a push/pull sync with an other agent.                                                                                                                                                                                                                                                                                                                                            | push/pull initiated / interval          | counter |
| `consul.memberlist.tcp.sent`          | This metric measures the total number of bytes sent by an agent through the TCP protocol                                                                                                                                                                                                                                                                                                                                                       | bytes sent / interval                   | counter |

> | `consul.memberlist.gossip`            | This metric measures the time taken for gossip messages to be broadcasted to a set of randomly selected nodes.                                                                                                                                                                                                                                                                                                                                 | messages / Interval                     | counter |

Reviewer: ```suggestion
| `consul.memberlist.gossip`            | This metric measures the time taken for gossip messages to be broadcasted to a set of randomly selected nodes.                                                                                                                                                                                                                                                                                                                                 | ms                  | timer |
```

> | `consul.memberlist.msg_alive`         | This metric counts the number of alive messages, that the agent has processed so far, based on the message information given by the network layer.                                                                                                                                                                                                                                                                                              | nodes / Interval                        | counter |
> | `consul.memberlist.msg_dead`          | This metric gives the number of dead messages, that the agent has processed so far, based on the message information given by the network layer.                                                                                                                                                                                                                                                                                                | nodes / Interval                        | counter |

Reviewer: ```suggestion
| `consul.memberlist.msg_alive`         | This metric counts the number of alive messages, that the agent has processed so far, based on the message information given by the network layer.                                                                                                                                                                                                                                                                                              | messages / Interval                        | counter |
| `consul.memberlist.msg_dead`          | This metric gives the number of dead messages, that the agent has processed so far, based on the message information given by the network layer.                                                                                                                                                                                                                                                                                                | messages / Interval                        | counter |
```

> | `consul.memberlist.msg_suspect`       | This metric gives the number of suspect messages, that the agent has processed so far, based on the message information given by the network layer.                                                                                                                                                                                                                                                                                              | nodes / Interval                        | counter |

Reviewer: ```suggestion
| `consul.memberlist.msg_suspect`       | This metric gives the number of suspect messages, that the agent has processed so far, based on the message information given by the network layer.                                                                                                                                                                                                                                                                                              | messages / Interval                        | counter |
```
